src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -576,11 +578,12 @@ static ImmutableArray<FunctionPointerParameterSymbol> makeParametersFromMetadata
                 }
             }
 
-            static RefKind getRefKind(ParamInfo<TypeSymbol> param, ImmutableArray<CustomModifier> paramRefCustomMods, RefKind hasInRefKind, RefKind hasOutRefKind)
+            static RefKind getRefKind(ParamInfo<TypeSymbol> param, ImmutableArray<CustomModifier> paramRefCustomMods, RefKind hasInRefKind, RefKind hasOutRefKind, bool requiresLocationAllowed)
             {
                 return param.IsByRef switch
                 {
                     false => RefKind.None,
+                    true when requiresLocationAllowed && CustomModifierUtils.HasRequiresLocationAttributeModifier(paramRefCustomMods) => RefKind.RefReadOnlyParameter,
                     true when CustomModifierUtils.HasInAttributeModifier(paramRefCustomMods) => hasInRefKind,
                     true when CustomModifierUtils.HasOutAttributeModifier(paramRefCustomMods) => hasOutRefKind,
```

Comment:
> they cannot ignore required modifiers
The compiler reports errors for invalid requires modifiers, there are tests for it, for example `FunctionPointer_ModreqRequiresLocation`. That happens elsewhere. This code only reads modifiers to determine the ref kind.

It's true that currently a `modreq(In) modopt(RequiresLocation)` parameter is recognized as `ref readonly`. There's a test for it - `FunctionPointer_ModreqIn_ModoptRequiresLocation`. Are you suggesting it should be an error instead? Note that the `modreq(In)` isn't actually unrecognized (if it was, it would be rejected) - although perhaps combining it with `modopt(RequiresLocation)` is unrecognized.

Reply:
OK. Disabled combining `modopt(RequiresLocation)` with other `modreq`s, don't want to change existing behavior here.